docs/design/SYSTEM_DESIGN.md

```diff
@@ -6,101 +6,124 @@
-Envoy Gateway consumes this configuration and provisions a unique fleet of Envoy Proxies. the [GatawayClass parameters](https://gateway-api.sigs.k8s.io/v1alpha2/api-types/gatewayclass/#gatewayclass-parameters) section allows the infrastructure administrator to further modify attributes of the data plane. 
-3. They will configure a [Gateway resource](https://gateway-api.sigs.k8s.io/concepts/api-overview/#gateway) linking it to a specific GatewayClass 
-with information such as hostnames, protocol and ports stating which traffic flows are of interest.
-4. Application developers can now expose their APIs by configuring [HTTPRoute resources](https://gateway-api.sigs.k8s.io/concepts/api-overview/#httproute).
-These routes are attached to a specific [Gateway resource](https://gateway-api.sigs.k8s.io/concepts/api-overview/#gateway) allowing application traffic to reach
-the upstream application.
-
-### Components
+#### Static Configuration
+This is the configuration used to configure various internal aspects of Envoy Gateway at runtime. Currently, Envoy
+Gateway only supports runtime configuration through a configuration file. See the [configuration](../CONFIG.md) guide
+for additional details.
 
-#### Config Sources
-This component is responsible for consuming the user configuration from various platforms. Data persistence should be tied to the specific config source’s capabilities. For e.g. in Kubernetes, the resources will persist in `etcd`, if using the `path-watcher`, the resources will persist in a file.
+#### Dynamic Configuration
+Dynamic configuration manages the data plane through [Gateway API][gw_api] objects, e.g. Gateway, HTTPRoute, etc. and
+provides the following services:
+* Infrastructure Management- Capabilities to manage, e.g. deploy, upgrade, etc. the data plane infrastructure. This is
+  expressed using [GatewayClass][gc] and [Gateway][gw] resources.
+* Traffic Routing- Capabilities that define how to handle application-level requests to backend services. This is
+  expressed using [HTTPRoute][hroute] and [TLSRoute][troute].
 
-##### Kubernetes
-It watches the Kubernetes API Server for resources, fetches them, and publishes it to the translators for further processing.
+#### Configuration Workflow
+See the [configuration](../CONFIG.md) guide for additional details on how to configure and run Envoy Gateway.
 
-##### Path Watcher
-It watches for file changes in a path, allowing the user to configure Envoy Gateway using resource configurations saved in a file or directory.
+### Components
 
-##### Config Server
-This is a HTTP/gRPC Server allowing Envoy Gateway to be configured from a remote endpoint. 
+#### Providers
+Providers are infrastructure components that Envoy Gateway calls to establish its
+[dynamic configuration](#dynamic-configuration), resolve services, persist data, and provide additional services through
```

Comment:
this sgtm, shouldnt the diagram reflect this by also encapsulating the service resolver as part of the provider ?